src/NuGet.Jobs.GitHubIndexer/ReposIndexer.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using NuGetGallery;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class ReposIndexer
+    {
+        private const string WorkingDirectory = "workdir";
+        private static readonly string GitHubUsageFilePath = WorkingDirectory + Path.DirectorySeparatorChar + "GitHubUsage.v1.json";
+        public static readonly string ExecutionDirectory = WorkingDirectory + Path.DirectorySeparatorChar + "exec";
+
+        private readonly IGitRepoSearcher _searcher;
+        private readonly ILogger<ReposIndexer> _logger;
+        private readonly int _maxDegreeOfParallelism;
+        private readonly IRepositoriesCache _repoCache;
+        private readonly IRepoFetcher _repoFetcher;
+        private readonly IConfigFileParser _configFileParser;
+
+        public ReposIndexer(IGitRepoSearcher searcher,
```

Comment:
```suggestion
        public ReposIndexer(
```

(new line before first arg)